linode/manager
# Automate change log script POC

Previous branch got too far behind. Sample gerenerate_changelog.py output:
`## [0.42.0] - 2018.12.17

### Added:
- Tag links (#4195)
- updating back up data section for dark theme (#4199)
- better padding for dashboard table cells
-  switch volume columns (#4197)
- add search data (#4182)
- Upgrade Typography component consistent with @material-ui/core@3.5.1 (#4180)
- Display resize instructions on form submission. (#4189)
- Add SSH key event message... (#4190)
- Refresh volumes list on volume_clone event. (#4191)
- Report counts of successes and failures for backups (#4187)
- Remove sendToast for enqueSnacback (#4185)
- Replace Toasts with Notistack (#4163)

### Changed:
- Events polling updated. (#4181)

### Fixed:
- Fix NodeBalancer Tag Error (#4194)
- Fix mutation error handling (#4179)`

**Purpose**
A python script that is more flexible than git auto changelog, searches for a list of keywords in the commit title rather than strict add: , fix: ect., also the script will not consider add tests or other automation commits to changelog. Additionally, a script that can generate a release candidate, bump package.json version, generate changelog, delete previous release branch, and push release candidate.

**To Test**
To test just the changelog script run python `~/manager/generate_changelog.py #.##.# yyyy.MM.dd origin` the first parameter is the release version, second date, third name of your manager remote repo. Then check for changelog updates.

@Jskobos used pull request template as discussed in tech talk 
@johnwcallahan updated contributing mark down
@martinmckenna to address your comment in the other pr, python `in` is a string function similar to JavaScript `.includes()` so `'fix' in 'fixed'` is true
